src/bsp/src/posix/src/TCPUartMock.cpp

```diff
@@ -0,0 +1,101 @@
+#include "TCPUartMock.h"
+#include "BSPUtils.h"
+#include <BSP.h>
+#include <FreeRTOS.h>
+#include <FreeRTOSConfig.h>
+#include <bsp/BSPContainer.h>
+#include <cstring>
+#include <ros/ros.h>
+#include <task.h>
+
+void TCPUartMock_listenTask(void* param) { static_cast<TCPUartMock*>(param)->waitForClient(); }
+
+TCPUartMock::TCPUartMock(ILogger& logger) : m_logger(logger), m_hasClient(false), m_port(0) {}
+
+TCPUartMock::~TCPUartMock() { close(); }
+
+void TCPUartMock::openSocket(int port) {
+    if (port == 0) {
+        m_logger.log(LogLevel::Info, "UART TCP mock port set to 0. Not initializing server.");
+        return;
+    }
+
+    m_port = port;
+    int serverFd;
+
+    if ((serverFd = ::socket(AF_INET, SOCK_STREAM, 0)) == 0) {
+        m_logger.log(LogLevel::Error, "UART TCP mock socket creation failed");
+        return;
+    }
+    m_serverFd = serverFd;
+
+    m_address.sin_family = AF_INET;
+    m_address.sin_addr.s_addr = INADDR_ANY;
+    m_address.sin_port = htons(m_port);
+
+    m_addressLength = sizeof(m_address);
+
+    if (::bind(m_serverFd, (struct sockaddr*)&m_address, static_cast<socklen_t>(m_addressLength)) <
+        0) {
+        m_logger.log(LogLevel::Error, "UART TCP mock server binding failed");
+        return;
+    }
+
+    m_logger.log(LogLevel::Info, "UART TCP mock server waiting for client on port %d", m_port);
+
+    xTaskCreate(TCPUartMock_listenTask, "tcp_uart_mock_listen", configMINIMAL_STACK_SIZE * 100,
+                (void*)this, tskIDLE_PRIORITY + 1, NULL);
+}
+
+bool TCPUartMock::send(const uint8_t* buffer, uint16_t length) {
+    if (!m_hasClient) {
```

Comment:
is it possible to use the client file descriptor instead of using 2 states?  i.e check if the file descriptor is != -1 for example

Or even an std::optional<int> m_clienFd

Reply:
Changed for std::optional